src/lib/storage/value_segment.hpp

```diff
@@ -45,7 +45,9 @@ class ValueSegment : public AbstractSegment {
   size_t estimate_memory_usage() const final;
 
  protected:
-  // Implementation goes here
+  std::vector<T> _values;
+  std::vector<bool> _null_values;
```

Comment:
Maybe make this one optional and initialize it only when the segment is `nullable`? (You can check whether the segment is nullable with `_null_values.has_value()`)